mne/io/brainvision/brainvision.py

```diff
@@ -513,9 +535,40 @@ def _get_vhdr_info(vhdr_fname, eog, misc, scale, montage):
                 if lp_s:
                     info['lowpass'] = 1. / info['lowpass']
         else:
-            info['lowpass'] = np.min(np.array(lowpass, dtype=np.float))
-            warn('Channels contain different lowpass filters. Lowest filter '
-                 'setting will be stored.')
+            heterogeneous_lp_filter = True
+            if lp_s:
+                # We convert channels with disabled filters to having
+                # infinitely relaxed / no filters
+                lowpass = [float(filt) if filt not in ('NaN', 'Off')
+                           else 0.0 for filt in lowpass]
+                info['lowpass'] = np.max(np.array(lowpass, dtype=np.float))
+                try:
+                    info['lowpass'] = 1. / info['lowpass']
+                except ZeroDivisionError:
+                    # No lowpass actually set (always Off or equivalent)
+                    # so we let lowpass be set in _empty_info
+                    del info['lowpass']
+                    # not actually heterogeneous in effect
+                    # ... just heterogeneously disabled
+                    heterogeneous_lp_filter = False
+            else:
+                # We convert channels with disabled filters to having
+                # infinitely relaxed / no filters
+                lowpass = [float(filt) if filt not in ('NaN', 'Off')
+                           else np.Inf for filt in lowpass]
+                info['lowpass'] = np.min(np.array(lowpass, dtype=np.float))
+
+                if np.isinf(info['lowpass']):
+                    # No lowpass actually set (always Off or equivalent)
+                    # so we let lowpass be set in _empty_info
+                    del info['lowpass']
+                    # not actually heterogeneous in effect
+                    # ... just heterogeneously disabled
+                    heterogeneous_lp_filter = False
+
+            if heterogeneous_lp_filter:
+                warn('Channels contain different lowpass filters. '
+                     'Lowest filter setting will be stored.')
```

Comment:
BV allows for different filters on different channels and the warning (again, carried forward from fairly old code) was because FIFF only supports global filters.